src/extension.ts

```diff
@@ -33,6 +35,8 @@ import { DefaultTelemetryService } from './shared/telemetry/defaultTelemetryServ
 import { ExtensionDisposableFiles } from './shared/utilities/disposableFiles'
 import { PromiseSharer } from './shared/utilities/promiseUtilities'
 
+const toolkitOutputChannel = getOutputChannel(OutputChannelName.ToolKit)
```

Comment:
Is it safe to call `getOutputChannel` before `nls.config`?

Reply:
Good call. It seems like that would be an issue. In our case we only _currently_  have one language so there is no difference at this time. I'm also not a fan of the cut and paste in every file:

```
import * as vscode from 'vscode'
import * as nls from 'vscode-nls'
const localize = nls.loadMessageBundle()
```

Perhaps the NLS initialization should happen in vsCodeUtils.ts and simply export `localize`, `OutputChannelName` and `getOutputChannel`.

Reply:
I think this portion of extension activation has to take place before calling `loadMessageBundle()` 

```
    const env = process.env as EnvironmentVariables
    if (!!env.VSCODE_NLS_CONFIG) {
        nls.config(JSON.parse(env.VSCODE_NLS_CONFIG) as nls.Options)()
    } else {
        nls.config()()
    }
```